crnk-core/src/main/java/io/crnk/core/engine/internal/utils/ClassUtils.java

```diff
@@ -395,4 +427,27 @@ public static Type getElementType(Type genericType) {
 		}
 		return genericType;
 	}
+
+	/**
+	 * Get the underlying class for a type, or null if the type is a variable type.
+	 * @param type the type
+	 * @return the underlying class
+	 */
+	private static Class<?> getRawTypeNullable(Type type) {
```

Comment:
you could make use of the Optional to make it explicit that there might be no raw type. Or should it be Object.class?

Reply:
I wrote this because the `getRawType` throws on `TypeVariable`s, which I think is not sensible. And it just blew up everything. The idea was that a type variable itself just does not have a raw type. With this semantics, I would prefer `Optional.empty()`, i.e. no raw type.

However, there might be a way out: If we would resolve the concrete type for that type variable, then we could very well get its raw type, and I think that would make sense. Maybe I'll do that afterwards.

Reply:
feel free to deprecate the current method and introduce a new one. Or even better rely on the BeanInformation where possible.